content/ADR-189-social-service-M2.md

```diff
@@ -0,0 +1,131 @@
+
+Now, when a user logs in to decentraland, the friendship requests will also be obtained from the social service instead of the matrix sdk.
+
+First, the client will need to login via HTTP, then establish a connection with the server, once correctly established, the server will then send two messages one with the current friends and another with the pending friend requests.
+
+```mermaid
+sequenceDiagram
+    participant User 1
+    participant Social service
+    participant Matrix
+
+    User 1 ->>+ Social service: Login
+    Social service ->>+ Matrix: Login
+    Matrix -->>- Social service: bearer token
+    Social service -->>- User 1: bearer token
+    User 1 ->>+ Social service: Connect (token)
+    Social service -->> User 1: Friends
+    Social service -->> User 1: Friend Requests
+    loop  Connection messages examples
+    User 1 ->> Social service: Friend event (ACCEPT, DELETE, etc.)
+    User 1 ->> Social service: Update position
+    Social service ->> User 1: Friend event (ACCEPT, DELETE, etc.)
+    Social service ->> User 1: Friend X new position
+    end
+    User 1 ->> Social service: Disconnect
+    Social service -->>- User 1: Disconnect
+```
+
+### Implementation
+
+Given that Browser interface is going to be slowly migrated into unity's codebase directly, the implementation of this feature will be done directly on unity's side directly too. Since this feature includes friends and friend requests, all of the code needed to handle friends and friendship requests will also be moved to unity, decoupling the dependency on Browser interface. Still, channels, presence, and messaging will be on the Browser interface side, but within time they will also be migrated to unity and will be able to leverage the new communication system.
+
+To prevent migrating all the code into unity renderer's code and adding a lot of logic, this ADR proposes to add a new library in C# that will handle of the logic around friendships and modify the logic in unity renderer to become more of an orchestrator of events. An example of a difference between current implementation and the proposal is the following:
+
+#### Current:
+```mermaid
+sequenceDiagram
+    participant Renderer
+    participant Browser interface
+    participant Matrix client
+
+    Renderer ->>+ Browser interface: Get friends
+    Browser interface ->> Matrix client: Get friends
+    Matrix client -->> Browser interface: friends ids []
+    Browser interface ->> Browser interface: Calculate missing profiles 
+    Browser interface ->> Renderer: Add profiles to catalog(Profiles [])
+    Browser interface -->>- Renderer: Friends []
+```
+
+
+#### Proposal:
+
+```mermaid
+sequenceDiagram
+    participant Renderer
+    participant Social client
+    participant Social service
+    participant Browser interface
+
+    Renderer ->>+ Social client: Get friends
+    Social client ->>+ Social service: Get friends
```

Comment:
Make explicit this is the _new_ Social Client in C# and not the current one in TS.